### PR TITLE
feat: support konflux banner

### PR DESCRIPTION
**Pull request by testcara**
## Fixes 
https://issues.redhat.com/browse/KFLUXUI-507

## Description
The konflux banner would be shown properly when the konflux banner content is there.
Both one-time banner and weekly/monthly banners are supported.

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 
i tested one time banner and monthly & weekly banner here and also tested some negative scenarios. 

https://github.com/user-attachments/assets/baaa7654-b089-4342-8486-e2b79280c3a1

If we enjoy markdown in the summary, the banner would be shown as:
Screenshot 2025-07-08 at 14-21-20 Overview Konflux

BTW, for local testing propose, you can specify the userBanner.ts to get the konflux-banner-configmap with you own tenant and create one configmap in your tenant like:
```
apiVersion: v1
data:
  banner-content.yaml: |
    enable: false
    type: danger # options: info, warning, danger
    summary: We’re investigating an issue causing intermittent build. Your workflows may experience slower execution. We’ll provide updates as we work to resolve the problem. Thank you for your patience.
kind: ConfigMap
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","data":{"banner-content.yaml":"enable: false\ntype: warning # options: info, warning, danger\nsummary: The Konflux platform will undergo scheduled maintenance tonight. Temporary service interruptions may occur.\n"},"kind":"ConfigMap","metadata":{"annotations":{},"name":"konflux-banner-configmap","namespace":"wlin-tenant"}}
  creationTimestamp: "2025-06-30T09:37:38Z"
  name: konflux-banner-configmap
  namespace: wlin-tenant
  resourceVersion: "2057177653"
  uid: 62a7b4d7-724b-4ba4-b6b2-aa55dcb92491

``` 
or more rich  one:
```
apiVersion: v1
data:
  banner-content.yaml: "enable: true\ntype: danger # options: info, warning, danger\nsummary:
    We’re investigating an issue causing intermittent build. <Strong>Your workflows
    may experience slower execution</Strong>. We’ll provide updates as we work to
    resolve the problem. See the [slack channel](https://redhat-internal.slack.com/archives/C06EHC5S0CS/p1751883098086219).\nrepeatType:
    weekly\ndayOfWeek: 2 \nstartTime: 03:00\nendTime: 23:59\n"
kind: ConfigMap
metadata:
  name: konflux-banner-configmap
  namespace: wlin-tenant
```

## How to test or reproduce?
provide different banner configmaps and see whether the banner is shown properly.

## Browser conformance: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/hooks/useBanner.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const parsed = yaml.load(yamlContent, { schema: yaml.FAILSAFE_SCHEMA });
+  if (typeof parsed !== 'object' || parsed === null) return null;
+  try {
+    const banner = bannerConfigYupSchema.validateSync(parsed, { abortEarly: false });
+    return {
+      ...banner,
+      summary: DOMPurify.sanitize(banner.summary),
+    };
+  } catch {
+    return null;
```
Comment (StanislavJochman) [inline src/hooks/useBanner.ts:29]: Can you do at least console log here?

Comment (StanislavJochman) [inline src/hooks/useBanner.ts:29]: Can you do at least console log that it failed here?

**Review by StanislavJochman — CHANGES_REQUESTED** [review]

File: src/hooks/useBanner.ts (hunk 1 above)
Comment (testcara) [reply src/hooks/useBanner.ts:31]: resolved. thank you.:-).

**Review by rrosatti — DISMISSED** [review]
I tested locally with my own tenant namespace and it's working fine! I left some minor comments, but I'm already leaving my approval :) nice one!

File: src/hooks/useBanner.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+export type BannerConfig = {
+  enable: boolean;
+  summary: string;
+  type: BannerType;
+  startTime?: string;
+  endTime?: string;
+};
+
+const parseBanner = (yamlContent: string): BannerConfig | null => {
+  const parsed = yaml.load(yamlContent, { schema: yaml.FAILSAFE_SCHEMA });
```
[the remark this answers is not in the document]
Comment (testcara) [reply src/hooks/useBanner.ts:20]: Thanks for raising this!
I currently use pure YAML in the ConfigMap for several reasons:
- YAML supports comments, which helps document the meaning of each field and improves maintainability.
-  YAML is widely used in Kubernetes ecosystem and is familiar to our operators.

The structure of the banner config is a bit complex , and YAML’s flexibility helps express that clearly and human readable for anyone  who would like to set banners.

In our codes, YAML is pretty easy to convert as json and do not bring unexpected complexity . 

So, I would like to keep yaml and leave the easiest way to any possible users.

File: src/hooks/useBanner.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    if (isLoading || error) return null;
+    const yamlContent = bannerYamlData?.data?.['banner-content.yaml'];
+    if (typeof yamlContent !== 'string') return null;
+    const parsedBanner = parseBanner(yamlContent);
+    if (!parsedBanner) return null;
+    const now = dayjs();
+    const show =
+      parsedBanner.enable &&
+      (!parsedBanner.startTime || now.isAfter(dayjs(parsedBanner.startTime))) &&
+      (!parsedBanner.endTime || now.isBefore(dayjs(parsedBanner.endTime)));
```
[the remark this answers is not in the document]
Comment (testcara) [reply src/hooks/useBanner.ts:59]: would update to support repeated banners.

File: src/hooks/useBanner.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+};
+
+const parseBanner = (yamlContent: string): BannerConfig | null => {
+  const parsed = yaml.load(yamlContent, { schema: yaml.FAILSAFE_SCHEMA });
+  if (typeof parsed !== 'object' || parsed === null) return null;
+  try {
+    const banner = bannerConfigYupSchema.validateSync(parsed, { abortEarly: false });
+    return {
+      ...banner,
+      summary: DOMPurify.sanitize(banner.summary),
```
[the remark this answers is not in the document]
Comment (testcara) [reply src/hooks/useBanner.ts:26]: my intention is to sanitize a html to ensure the fronted can render it safely.
Would support markdown and keep the sanitize here.

File: src/hooks/useBanner.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+export const useBanner = () => {
+  const {
+    data: bannerYamlData,
+    isLoading,
+    error,
+  } = useK8sWatchResource<ConfigMap>(
+    {
+      groupVersionKind: ConfigMapGroupVersionKind,
+      namespace: 'konflux-info',
+      name: 'konflux-banner-configmap',
```
[the remark this answers is not in the document]
Comment (testcara) [reply src/hooks/useBanner.ts:107]: updated

File: src/components/konfluxBanner/KonfluxBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  return (
+    <Banner isSticky variant={typeToVariant(banner.type)} data-test="banner">
+      <Flex
+        alignItems={{ default: 'alignItemsCenter' }}
+        justifyContent={{ default: 'justifyContentCenter' }}
+        flexWrap={{ default: 'nowrap' }}
+        style={{ width: '100%' }}
+      >
+        <FlexItem>
+          {typeToIcon(banner.type)} {banner.summary}
```
[the remark this answers is not in the document]
Comment (testcara) [reply src/components/konfluxBanner/KonfluxBanner.tsx]: updated

File: src/hooks/useBanner.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (testcara) [reply src/hooks/useBanner.ts]: updated

File: src/hooks/useBanner.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (testcara) [reply src/hooks/useBanner.ts]: updated
